data/SugarBean.php

```diff
@@ -3629,22 +3629,24 @@ public function create_new_list_query(
         //walk through the fields and for every relationship field add their relationship_info field
         //relationshipfield-aliases are resolved in SugarBean::create_new_list_query
         // through their relationship_info field
-        $addrelate = array();
-        foreach ($fields as $field => $value) {
-            if (isset($this->field_defs[$field]) && isset($this->field_defs[$field]['source']) &&
-                $this->field_defs[$field]['source'] == 'non-db'
-            ) {
-                $addrelatefield = $this->get_relationship_field($field);
-                if ($addrelatefield) {
-                    $addrelate[$addrelatefield] = true;
+        if (isset($parentbean)) {
+            $addrelate = array();
+            foreach ($fields as $field => $value) {
+                if (isset($this->field_defs[$field]) && isset($this->field_defs[$field]['source']) &&
+                    $this->field_defs[$field]['source'] == 'non-db'
+                ) {
+                    $addrelatefield = $this->get_relationship_field($field, $parentbean->module_name);
+                    if ($addrelatefield) {
+                        $addrelate[$addrelatefield] = true;
+                    }
+                }
+                if (!empty($this->field_defs[$field]['id_name'])) {
+                    $addrelate[$this->field_defs[$field]['id_name']] = true;
                 }
             }
-            if (!empty($this->field_defs[$field]['id_name'])) {
-                $addrelate[$this->field_defs[$field]['id_name']] = true;
-            }
-        }
 
-        $fields = array_merge($addrelate, $fields);
+            $fields = array_merge($addrelate, $fields);
+        }
```

Comment:
Check if $parentbean is set, which is a requirement to lookup relationship_info fields, and pass $parentbean->module_name on to the method get_relationship_field.